std/range/package.d

```diff
@@ -7125,9 +7131,57 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Chunks* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
+        @property auto front() { return impl.r.front; }
+        void popFront()
+        {
+            assert(impl.curSizeLeft > 0 && !impl.r.empty);
+            impl.curSizeLeft--;
+            impl.r.popFront();
+        }
+    }
+
+    private Source r;
+    private size_t chunkSize;
+    private size_t curSizeLeft;
+    bool empty;
+
+    this(Source _r, size_t _chunkSize)
```

Comment:
I think these parameter are publicly exposed, so I think it's better to use `this.x = x` below (even if it's a bit uglier).